tests/matrix.rs

```diff
@@ -365,3 +365,29 @@ fn test_from_angle() {
     let rot3: Matrix2<f64> = Matrix2::from_angle(rad(f64::consts::PI));
     assert!((rot3 * Vector2::new(1.0, 1.0)).approx_eq(&Vector2::new(-1.0, -1.0)));
 }
+
+#[test]
+fn test_matrix3_into_quaternion()
+{
+    fn eq(a: Matrix3<f32>, b: Matrix3<f32>) {
+        if !a.approx_eq(&b) {
+            panic!("{:?} != {:?}", a, b);
+        }
+    }
+
+    fn test_angles(x: Rad<f32>, y: Rad<f32>, z: Rad<f32>) {
+        let basis: Basis3<f32> = Rotation3::from_euler(x, y, z);
+        let quaternion: Quaternion<f32> = basis.into();
+        eq(basis.into(), quaternion.into());
+    }
+
+    // trigger each of the cases in: from(mat: Matrix3<S>) -> Quaternion<S>
+    // trigger the case: trace >= S::zero()
+    test_angles(rad(0.0f32), rad(0.0), rad(0.0f32));
+    // trigger the case: (mat[0][0] > mat[1][1]) && (mat[0][0] > mat[2][2])
+    test_angles(rad(2.0f32), rad(1.0), rad(-1.2f32));
+    // trigger the case: (mat[0][0] > mat[1][1]) && (mat[0][0] > mat[2][2])
+    test_angles(rad(2.0f32), rad(1.0), rad(3.0f32));
+    // trigger the else case
+    test_angles(rad(1.0f32), rad(1.0), rad(3.0f32));
```

Comment:
Would be nice if these were separate tests so that we get more fine-grained failures.